macro/src/lib.rs

```diff
@@ -94,6 +94,21 @@ fn gen_params_at_boundary(param: &ast::Param, expanded_params: &mut Vec<FnArg>)
                 ),
             }));
         }
+        ast::TypeName::Function(..) => {
```

Comment:
issue: I would like to have some snapshot tests at the bottom of this file that show the generated code for callback stuff

Reply:
yes, great point.